pkg/nodeconfig/nodeconfig.go

```diff
@@ -193,13 +192,14 @@ func (nc *nodeConfig) Configure() error {
 			return errors.Wrap(err, "configuring WICD failed")
 		}
 		// Set the desired version annotation, communicating to WICD which Windows services configmap to use
-		if err := nc.applyLabelsAndAnnotations(nil, map[string]string{DesiredVersionAnnotation: version.Get()}); err != nil {
+		if err := metadata.ApplyDesiredVersionAnnotation(context.TODO(), nc.client, *nc.node, wmcoVersion); err != nil {
 			return errors.Wrapf(err, "error updating desired version annotation on node %s", nc.node.GetName())
 		}
 
-		// Now that basic kubelet configuration is complete, configure networking in the node
-		if err := nc.configureNetwork(); err != nil {
-			return errors.Wrap(err, "configuring node network failed")
+		// Wait for version annotation. This prevents uncordoning the node until all node services and networks are up
+		if err := metadata.WaitForVersionAnnotation(context.TODO(), nc.client, *nc.node, wmcoVersion); err != nil {
+			return errors.Wrapf(err, "error waiting for proper %s annotation for node %s", metadata.VersionAnnotation,
+				nc.node.GetName())
 		}
 
 		// Now that the node has been fully configured, add the version annotation to signify that the node
```

Comment:
Outdated comment and potentially outdated logic

Reply:
Adjusted the comment. The logic is still fine as we want to update the nc.node object after applying all these annotations to prevent stale resource